tensornetwork/matrixproductstates/mps_class_test.py

```diff
@@ -0,0 +1,204 @@
+  ] + [np.random.randn(D, d, 1)]
+  with pytest.raises(ValueError):
+    tensornetwork.mps.FiniteMPS(tensors, center_position=pos, backend=backend)
+
+
+@pytest.mark.parametrize("backend,dtype", [('numpy', np.float64),
+                                           ('numpy', np.complex128),
+                                           ('tensorflow', np.float64),
+                                           ('tensorflow', np.complex128),
+                                           ('pytorch', np.float64),
+                                           ('jax', np.float64)])
+def test_left_orthonormalization(backend, dtype):
+  D, d, N = 10, 2, 10
+  tensors = [get_random_np((1, d, D), dtype)] + [
+      get_random_np((D, d, D), dtype) for _ in range(N - 2)
+  ] + [get_random_np((D, d, 1), dtype)]
+  mps = tensornetwork.mps.FiniteMPS(
+      tensors, center_position=N - 1, backend=backend)
+  mps.position(0)
+  mps.position(len(mps) - 1)
+  assert all([
+      mps.check_orthonormality('left', site) < 1E-12
+      for site in range(len(mps))
+  ])
+
+
+@pytest.mark.parametrize("backend,dtype", [('numpy', np.float64),
+                                           ('numpy', np.complex128),
+                                           ('tensorflow', np.float64),
+                                           ('tensorflow', np.complex128),
+                                           ('pytorch', np.float64),
+                                           ('jax', np.float64)])
+def test_right_orthonormalization(backend, dtype):
+  D, d, N = 10, 2, 10
+  tensors = [get_random_np((1, d, D), dtype)] + [
+      get_random_np((D, d, D), dtype) for _ in range(N - 2)
+  ] + [get_random_np((D, d, 1), dtype)]
+  mps = tensornetwork.mps.FiniteMPS(tensors, center_position=0, backend=backend)
+
+  mps.position(len(mps) - 1)
+  mps.position(0)
+  assert all([
+      mps.check_orthonormality('right', site) < 1E-12
+      for site in range(len(mps))
+  ])
+
+
+@pytest.mark.parametrize("backend,dtype", [('numpy', np.float64),
+                                           ('numpy', np.complex128),
+                                           ('tensorflow', np.float64),
+                                           ('tensorflow', np.complex128),
+                                           ('pytorch', np.float64),
+                                           ('jax', np.float64)])
+def test_apply_one_site_gate(backend, dtype):
+  D, d, N = 10, 2, 10
+  tensors = [get_random_np((1, d, D), dtype)] + [
+      get_random_np((D, d, D), dtype) for _ in range(N - 2)
+  ] + [get_random_np((D, d, 1), dtype)]
+  mps = tensornetwork.mps.FiniteMPS(tensors, center_position=0, backend=backend)
+  gate = get_random_np((2, 2), dtype)
+  mps.apply_one_site_gate(gate, 5)
+  actual = np.transpose(np.tensordot(tensors[5], gate, ([1], [1])), (0, 2, 1))
+  np.testing.assert_allclose(mps.nodes[5].tensor, actual)
+
+
+@pytest.mark.parametrize("backend,dtype", [('numpy', np.float64),
```

Comment:
I think you can avoid copying and pasting these by using pytest's fixures: https://pybit.es/pytest-fixtures.html

We use this already here https://github.com/google/TensorNetwork/blob/50849ce792a1dcdf2a445af83125d06a5c25df09/conftest.py#L21, You could do the same by adding a local fixure in this file.

Reply:
Are you suggesting to add a fixture to conftest?

Reply:
btw, is that typo there on purpose?

Reply:
I'm not very familiar with this. where is `backend_fixure` called?

Reply:
No, you would add the fixture to this file and not conftest.py

And pytest just handles it magically. It's great!

Reply:
Allright, and how exactly do I define a fixture with the particular combination of backend-dtype arguments (not all backends are called for all dtypes)?

Reply:
@Thenerdstation added the fixture